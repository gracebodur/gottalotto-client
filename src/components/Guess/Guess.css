.guesslist-container section {
  display: flex;
  flex-direction: row;
  padding: 15px;
}

.guesslist-container {
  text-align: center;
  display: flex;
  justify-content: center;
  color: #fff;
  font-size: 30px;
  padding: 15px;
}
.flippy-card {
  z-index: 0;
}
