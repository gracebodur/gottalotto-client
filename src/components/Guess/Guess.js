import React, { Component } from 'react'
import avatar from '../../img/winner_avatar.png'
import Flippy, { FrontSide, BackSide } from 'react-flippy'
import './Guess.css'

class Guess extends Component {
	render() {
		const { user_name, guess_1, guess_2, guess_3, guess_4, guess_5, power_ball, message, guess_created_date, has_won } = this.props.guess
		return (
			<section className='guesslist-container'>
				<div style={{ display: 'flex', flex: '1 0 200px', justifyContent: 'space-around', 'flex-wrap': 'wrap' }}>
					<Flippy
						flipOnHover={true}
						flipDirection='vertical'
						style={{ width: '300px', height: '400px' }}
					>
						<FrontSide style={{
							backgroundColor: '#41669d',
							display: 'flex',
							alignItems: 'center',
							flexDirection: 'column'
						}} >
							<img src={avatar} alt='avatar' style={{ maxWidth: '150px', maxHeight: '150px' }} />
							{user_name}

							<span
								className='userName'
								style={{
									fontSize: '17px',
									position: 'absolute',
									bottom: '10px',
									width: '100%',
								}}>
								Guessed Numbers:<br /><br />
								{guess_1}, {guess_2}, {guess_3}, {guess_4}, {guess_5}, PB: {power_ball}<br />
							</span>
						</FrontSide>
						<BackSide
							style={{
								backgroundColor: '#175852',
								display: 'flex',
								alignItems: 'center',
								justifyContent: 'center',
								flexDirection: 'column'
							}}>
							{message}
							<span className='guesslistNumbers'
								style={{
									fontSize: '17px',
									position: 'absolute',
									bottom: '10px',
									width: '100%'
								}}>
							</span>
						</BackSide>
					</Flippy>
				</div>
			</section>
		)
	}
}

export default Guess


