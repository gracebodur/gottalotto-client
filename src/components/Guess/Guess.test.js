import React from "react";
import toJson from "enzyme-to-json";
import Guess from "./Guess";

describe(`Guess component`, () => {
	const props = {
		id: "a",
		name: "test-class-name"
	};

	it("renders a Guess by default", () => {
		const wrapper = <Guess />;
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it("renders the Guess given props", () => {
		const wrapper = <Guess {...props} />;
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});