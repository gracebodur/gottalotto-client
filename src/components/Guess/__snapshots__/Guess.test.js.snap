// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Guess component renders a Guess by default 1`] = `undefined`;

exports[`Guess component renders the Guess given props 1`] = `undefined`;
